raytracing/matrix.py

```diff
@@ -1071,6 +1071,14 @@ def pointsOfInterest(self, z):
 
         return pointsOfInterest
 
+    def transferMatrix(self, upTo=float('+Inf')):
+        """ Returns a ThickLens() corresponding to a partial propagation
+        if the requested distance is smaller than the length of this element"""
+        if self.L <= upTo:
+            return self
+        else:
+            return ThickLens(self.n, self.R1, self.R2, upTo, self.apertureDiameter, self.label)
```

Comment:
You need to return a Matrix (or MatrixGroup) that has DielectricInterface + Space(appropriate length).